File: navcodes/recursion/1_reverse_a_string.py
def reverse_string(s):
    if not s:
        # base case or stopping condition
        return "" 
    # what is the smallest amount of work I can do in each iteration?
    return reverse_string(s[1:]) + s[0]


if __name__ == "__main__":
    print(reverse_string("Hello World"))

File: recursion_js/1_reverse_a_string.js
function reverseString(s) {
    if (s.length === 0 || s.length === 1) {
        return s; // base case: string of length 0 or 1 is already reversed
    }
    // what is the smallest amount of work I can do in each iteration?
    return reverseString(s.slice(1)) + s[0];
}

s = reverseString("Hello World")
console.log(s)
